File: 1768.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        merged = []
        i, j = 0, 0
        
        # Loop through both words, alternating characters
        while i < len(word1) and j < len(word2):
            merged.append(word1[i])
            merged.append(word2[j])
            i += 1
            j += 1
        
        # Append remaining characters from the longer string
        if i < len(word1):
            merged.append(word1[i:])
        elif j < len(word2):
            merged.append(word2[j:])
        
        return ''.join(merged)

File: 1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder merged = new StringBuilder();
        int i = 0, j = 0;
        
        // Loop through both words, alternating characters
        while (i < word1.length() && j < word2.length()) {
            merged.append(word1.charAt(i++));
            merged.append(word2.charAt(j++));
        }
        
        // Append the remaining characters from the longer string
        if (i < word1.length()) {
            merged.append(word1.substring(i));
        } else if (j < word2.length()) {
            merged.append(word2.substring(j));
        }
        
        return merged.toString();
    }
}
